src/core/Akka/Routing/SmallestMailbox.cs

```diff
@@ -35,7 +35,7 @@ private Routee SelectNext(Routee[] routees)
             var winningScore = long.MaxValue;
 
             // round robin fallback
-            var winner = routees[Interlocked.Increment(ref next) % routees.Length];
+            var winner = routees[(Interlocked.Increment(ref next) & int.MaxValue) % routees.Length];
```

Comment:
Hmm.. I got the distinct impression that the Interlocked.Increment handled the overflow scenario internally.